packages/runtime-core/__tests__/components/Teleport.spec.ts

```diff
@@ -408,4 +408,31 @@ describe('renderer: teleport', () => {
       `"<div>teleported</div><span>false</span><!--v-if-->"`
     )
   })
+
+  test('should not throw error', async () => {
+    const targetA = nodeOps.createElement('div')
+    const targetB = nodeOps.createElement('div')
+    const target = ref(targetA)
+    const root = nodeOps.createElement('div')
+
+    const children = ref([h('div', 'teleported')])
+    render(
+      h(() => [
+        h(Teleport, { to: target.value }, children.value),
+        h('div', 'root')
+      ]),
+      root
+    )
+
+    try {
+      target.value = targetB
+      await nextTick()
+    } catch (err) {
+      expect(err).toBeUndefined()
```

Comment:
why not remove the try catch altogether, I don't think it makes sense for this code to throw an   undefined error

Reply:
Yeah, it looks a little weird. I'm just trying to show you that an error is thrown when you excute this test case before fixing it.
Actually, I wanted to refactor the test cases after you merged my PRs. Because this one can be tested together with the test case of "should update target" that I modified in my other PR.

Reply:
@leaon4 I'm not sure what other PR you are referring to, can you clarify?

Reply:
@LinusBorg #2870 #2871